### PR TITLE
Slimeperson customization 2022 expansion pack (also refactors a bit of the code related to it)

### PR DESCRIPTION
**Pull request by mcmeiler**
## About The Pull Request

Adds lizard tails, frills, spines and wings to slimeperson shapeshifting

Deleted most copy-paste code in favor of separate procs to handle the sprite assignment/removal

## Why It's Good For The Game

more variety = good, I think

## Changelog
:cl:
add: lizard tails, frills, spines and wings (non-functional, nice try, powergamer) to slimeperson shapeshifting
refactor: made slimeperson shapeshifting code *slightly* less awful.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -972,6 +972,9 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 					S = GLOB.kepori_body_feathers_list[H.dna.features["kepori_body_feathers"]]
 			if(!S || S.icon_state == "none")
 				continue
+			if(isjellyperson(H))
+				S.image_alpha = 150 //Jellypeople sprite handling
+				S.hasinner = FALSE
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/human/species.dm]: Snowflake shit

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm [hunk 2]
```diff
@@ -171,44 +171,97 @@
...
+			var/list/part_list = list("Ears", "Tails", "Frills", "Spines", "Wings") //Selection, if you want more stuff added, do it yourself
+			var/selected_part = input(owner, "Which bodypart do you want to alter?", "Mutant Bodypart Alteration") as null|anything in part_list + "Cancel"
+			if(selected_part)
+				if(selected_part == "Cancel")
+					return
+				switch(selected_part)
+					if("Ears")
+						handle_sprites(lowertext(selected_part), H, GLOB.ears_list)
+					if("Tails") //Tails are a special case since they're separate from each other, and go into different features and mutant_bodyparts.
+						var/datum/sprite_accessory/tails/SA = new(src) // I *could* move this to a separate proc, but I don't see the point. It'd only be called this one time.
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/human/species_types/jellypeople.dm]: dont instantiate an object, just use initial()

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm [hunk 3]
```diff
@@ -171,44 +171,97 @@
...
+							H.dna.species.mutant_bodyparts |= "tail_lizard"
+							H.dna.features["tail_lizard"] = SA.name
+							H.update_body()
+							qdel(SA)
+							return
+						if(SA.name in GLOB.tails_list_human)
+							H.dna.species.mutant_bodyparts |= "tail_human"
+							H.dna.features["tail_human"] = SA.name
+							H.update_body()
+							qdel(SA)
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/human/species_types/jellypeople.dm]: wow thats alot of copy paste

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm [hunk 4]
```diff
@@ -171,44 +171,97 @@
...
+		H.dna.features["[sprite]"] = SA.name
+		H.update_body()
+		qdel(SA)
+		return
+	else
+		H.dna.species.mutant_bodyparts |= "[sprite]"
+		H.dna.features["[sprite]"] = SA.name
+		H.update_body()
+		qdel(SA)
+		return
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/human/species_types/jellypeople.dm]: alot more copy paste. thin this out.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Comment by mcmeiler** [comment]
Go ahead and yell at me all you want, I'm going to bed. This still sucks, but it's no longer copy-paste.

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm [hunk 5]
```diff
@@ -171,44 +171,94 @@
...
-						H.update_body()
+
+				//Mutant Bodyparts
+
+		if("Mutant Bodyparts")
+			var/list/part_list = list("Ears", "Tails", "Frills", "Spines", "Wings") //Selection, if you want more stuff added, do it yourself
+			var/selected_part = input(owner, "Which bodypart do you want to alter?", "Mutant Bodypart Alteration") as null|anything in part_list + "Cancel"
+			if(selected_part)
+				if(selected_part == "Cancel")
+					return
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/human/species_types/jellypeople.dm:199]:
```suggestion
			var/static/list/part_list = list("Ears", "Tails", "Frills", "Spines", "Wings") //Selection, if you want more stuff added, do it yourself
			var/selected_part = input(owner, "Which bodypart do you want to alter?", "Mutant Bodypart Alteration") as null|anything in part_list + "Cancel"
			if(!selected_part || selected_part == "Cancel")
				return
```

File: code/modules/mob/living/carbon/human/species.dm [hunk 6]
```diff
@@ -972,6 +972,9 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 					S = GLOB.kepori_body_feathers_list[H.dna.features["kepori_body_feathers"]]
 			if(!S || S.icon_state == "none")
 				continue
+			S.image_alpha = H.dna.species.hair_alpha //hair_alpha is a decent indicator of how transparent a species is
+			if(isjellyperson(H))
+				S.hasinner = FALSE //Jellies shouldn't get inner sprites, where would they get the second color from?
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/human/species.dm:977]: this is still snowflakey and awful.

File: code/modules/mob/dead/new_player/sprite_accessories.dm [hunk 7]
```diff
@@ -1792,7 +1792,7 @@
 	hasinner = 1
 	color_src = HAIR
 
-/datum/sprite_accessory/ears/cat/slime
+/datum/sprite_accessory/ears/cat/slime //I would yeet these, but unfortunately, deleting these can mess up savefiles, so you're stuck with this :)
```
Comment (Kapu1178) [inline code/modules/mob/dead/new_player/sprite_accessories.dm:1795]: You can remove these. im pretty sure. Ask mark.

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm [hunk 8]
```diff
@@ -171,44 +171,94 @@
...
+						handle_sprites(lowertext(selected_part), H, GLOB.ears_list)
+					if("Tails") //Tails are special and need special treatment because they're a combination of 2 lists.
+						handle_tails(H)
+					if("Frills")
+						handle_sprites(lowertext(selected_part), H, GLOB.frills_list)
+					if("Spines")
+						handle_sprites(lowertext(selected_part), H, GLOB.spines_list)
+					if("Wings") //These don't look too good
+						handle_sprites(lowertext(selected_part), H, GLOB.wings_list)
+	return
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/human/species_types/jellypeople.dm:211]: Just dump an `update_body(H)` before the return and remove all the other update_body() calls

**Review by Kapu1178 — CHANGES_REQUESTED** [review]
These in addition to what was discussed on discord

File: code/modules/mob/dead/new_player/sprite_accessories.dm (hunk 7 above)
Comment (MarkSuckerberg) [reply code/modules/mob/dead/new_player/sprite_accessories.dm:1795]: Yeah, it'll just sanitize it and re-randomize if needed I believe.

**Pull request closed without merging**